src/evox/algorithms/so/de_variants/ode.py

```diff
@@ -146,6 +144,8 @@ def _ask_one(self, index, R, population, best_index, random_choiced, mask):
         return trial_vector
 
     def tell(self, state, trial_fitness):
+        if self.is_odd:
```

Comment:
`is_odd` belongs to `state`, not `self`. `self` only stores static data, so after the compilation, `self.is_odd` will always be True.
You can refer to the BCE-IBEA's implementation on how to apply different update strategies on odd/even generations. Basically, you need to store `is_odd` in the state and use `jax.lax.cond` to pick the correct code path.

Reply:
Thanks a lot. I'll rewrite the code about `is_odd` and check it again. 🫡

Reply:
😀

Reply:
I've modified the logic using `jax.lax.cond`, and the tests pass✔. Please check it.